### PR TITLE
Fix problems of previous PR

### PR DESCRIPTION
**Pull request by frapac**
Take into account issues raised during last PR
- Improve import of modules in examples
- update unit-testing
- add a dams valley management examples
- fix code in multi-dimensional state case

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -38,7 +38,7 @@ facts("Utility functions") do
     include("../src/utility.jl")
 
     V = get_null_value_functions()
-    @fact typeof(V) --> SDDP.PolyhedralFunction
+    @fact typeof(V) --> PolyhedralFunction
     @fact V.betas[1] --> 0
 
     arr = rand(4, 4, 2)
```
Comment (Henri-Gerard) [inline test/runtests.jl:44]: Ok

File: src/utility.jl [hunk 2]
```diff
@@ -74,9 +73,9 @@ Returns :
...
 
 """
-function upper_bound(model::SDDP.SPModel,
-                     param::SDDP.SDDPparameters,
-                     V::Vector{SDDP.PolyhedralFunction},
+function upper_bound(model::SPModel,
+                     param::SDDPparameters,
+                     V::Vector{PolyhedralFunction},
                      forwardPassNumber::Int64,
                      returnMCerror::Bool)
```
Comment (Henri-Gerard) [inline src/utility.jl:81]: Ok

File: src/oneStepOneAleaProblem.jl [hunk 3]
```diff
@@ -93,12 +94,11 @@ function solve_one_step_one_alea(model, #::SDDP.LinearDynamicLinearCostSPmodel,
...
-
         # Return object storing results:
         result = SDDP.NextStep(
                           [model.dynamics(xt, optimalControl, xi)],
                           optimalControl,
-                          getDual(m.ext[:cons][1]),
+                          [getDual(m.ext[:cons][i]) for i in 1:model.dimStates],
                           getObjectiveValue(m))
     else
         # If no solution is found, then return nothing
```
Comment (Henri-Gerard) [inline src/oneStepOneAleaProblem.jl:104]: I think that for the moment we can keep going with a fixed dimension of control and state but a TODO would be great :)

File: src/forwardBackwardIterations.jl [hunk 4]
```diff
@@ -238,6 +240,7 @@ function backward_pass(model, #::SDDP.SPModel,
                                                        1), 1)
                 add_constraints_with_cut!(model, solverProblems[t-1], V[t])
             else
+                subgradient = Array{Float64}(subgradient)
                 add_cut!(model, solverProblems[t-1], V[t], beta[1], subgradient)
             end
```
Comment (Henri-Gerard) [inline src/forwardBackwardIterations.jl:246]: Ok

File: src/SDDPoptimize.jl [hunk 5]
```diff
@@ -183,8 +188,8 @@ Returns :
     each value function
 
 """
-function optimize(model::SDDP.SPModel,
-                  param::SDDP.SDDPparameters,
+function optimize(model::SPModel,
+                  param::SDDPparameters,
                   n_iterations=20,
                   display=true)
```
Comment (Henri-Gerard) [inline src/SDDPoptimize.jl:195]: Ok

File: examples/dam.jl [hunk 6]
```diff
@@ -62,7 +62,7 @@ end
 in advance."""
 function solve_determinist_problem()
     println(alea_year)
-    m = Model(solver=CplexSolver(CPX_PARAM_SIMDISPLAY=0))
+    m = Model(solver=SOLVER)
 
     @defVar(m,  0           <= x[1:(TF+1)]  <= 100)
```
Comment (leclere) [inline examples/dam.jl:67]: Seems good to me. Would just prefer to see 7 and 100 to be constants defined at the beginning of the file.

**Pull request merged**